parser/parser.y

```diff
@@ -2561,6 +2562,10 @@ FunctionCallNonKeyword:
 		args := []ast.ExprNode{$3.(ast.ExprNode), $5.(ast.ExprNode), $7.(ast.ExprNode)}
 		$$ = &ast.FuncCallExpr{FnName: model.NewCIStr($1.(string)), Args: args}
 	}
+| "REVERSE" '(' Expression ')'
```

Comment:
reverse is an unreserved keyword. So please add it into https://github.com/pingcap/tidb/blob/master/parser/parser.y#L1908. Meanwhile add a test case here: https://github.com/pingcap/tidb/blob/master/parser/parser_test.go#L37